activerecord/lib/active_record/associations.rb

```diff
@@ -1294,7 +1294,7 @@ module ClassMethods
         #   * <tt>:destroy</tt> causes all the associated objects to also be destroyed.
         #   * <tt>:delete_all</tt> causes all the associated objects to be deleted directly from the database (so callbacks will not be executed).
         #   * <tt>:nullify</tt> causes the foreign keys to be set to +NULL+. Callbacks are not executed.
-        #   * <tt>:restrict_with_exception</tt> causes an exception to be raised if there are any associated records.
+        #   * <tt>:restrict_with_exception</tt> causes an <tt>ActiveRecord::DeleteRestrictionError</tt> exception to be raised if there are any associated records.
```

Comment:
Can you address L1440 too?

https://github.com/rails/rails/blob/d2c8e9b4efcbee5785dd4c54956254bdcfaa2b9c/activerecord/lib/active_record/associations.rb#L1440